java/core/src/main/java/com/microsoft/bond/protocol/UntaggedProtocolReader.java

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond.protocol;
+
+import java.io.IOException;
+
+/**
+ * Reads from serialized payload encoded using an untagged protocol.
+ */
+public interface UntaggedProtocolReader {
+    /**
+     * Start reading a container.
+     */
+    int readContainerBegin() throws IOException;
+
+    /**
+     * End reading a container.
+     */
+    void readContainerEnd() throws IOException;
+
+    /**
+     * Read an int8.
+     *
+     * @return the value
+     */
+    byte readInt8() throws IOException;
+
+    /**
+     * Skip an int8.
+     */
+    void skipInt8() throws IOException;
```

Comment:
Is this for Bonded?

---
In reply to: [130756104](https://github.com/Microsoft/bond/pull/542#discussion_r130756104) [](ancestors = 130756104)